docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,202 @@
+you run the script), the app will automatically initiate an auth flow to acquire new
+tokens. For a UserApp, this will print out a URL to terminal with a prompt instructing a
+the user to follow the link and enter the code they're given back into the terminal. For
+a ClientApp, the app will retrieve tokens programmatically through an Auth API.
+
+Once this auth flow has finished, the app will cache tokens for future use and
+invocation of your requested method will proceed as expected.
+
+
+Manually Running Login Flows
+----------------------------
+
+While your app will automatically initiate and oversee auth flows as detected, sometimes
+the programmer knows timing better. To manually trigger a login flow, call
+``GlobusApp.run_login_flow(...)``. This will initiate a flow requesting new tokens
+based on the app's currently defined scope requirements, caching the resulting tokens
+for future use.
+
+This method accepts a single optional parameter, ``auth_params``, where a caller
+may specify additional session-based auth parameters such as requiring the use of an
+mfa token or rendering with a specific message:
+
+
+..  code-block:: python
+
+    from globus_sdk.experimental.auth_requirements_error import GlobusAuthorizationParameters
+
+    ...
+
+    my_app.run_login_flow(
+        auth_params=GlobusAuthorizationParameters(
+            session_message="Please authenticate with MFA",
+            session_required_mfa=True,
+        )
+    )
+
+
+Manually Defining Scope Requirements
+------------------------------------
+
+Globus service clients all maintain as a part of their class definition a list of
+default scope requirements to be attached to any bound app. These scopes represent our
+best approximation of a "standard set" for this service; unfortunately however,
+this list will note be sufficient for all use cases.
+
+For example, the FlowsClient defines its default scopes as ``flows:view_flows`` and
+``flows:run_status`` (read-only access). These scopes will not be sufficient for a
+script which needs to create new flows or modify existing ones. For that script, the
+author must manually attach the ``flows:manage_flows`` scope to the app.
+
+This can be done in one of two ways:
+
+#.  Through a service client initialization, using the ``app_scopes`` kwarg.
+
+    ..  code-block:: python
+
+        from globus_sdk import Scope, FlowsClient
+
+        FlowsClient(app=my_app, app_scopes=[Scope(FlowsClient.scopes.manage_flows)])
+
+    This approach results in an app which only requires the ``flows:manage_flows``
+    scope. Neither default scope (``flows:view_flows`` and ``flows:run_status``) are
+    registered.
+
+#.  Through a service client's ``add_app_scope`` method.
+
+    ..  code-block:: python
+
+        from globus_sdk import Scope, FlowsClient
+
+        flows_client = FlowsClient(app=my_app)
+        flows_client.add_app_scope(Scope(FlowsClient.scopes.manage_flows))
```

Comment:
It feels very unnecessary that the "clearly already a scope" thing, which is being passed to something that "clearly wants a scope", has to be wrapped in a `Scope` object.

Is this strictly necessary today? Could it instead be:

```suggestion
        flows_client.add_app_scope(FlowsClient.scopes.manage_flows)
```

If so, that's what I want to see here.